For some reason mtoc assigns IMAGE_SCN_CNT_CODE and
IMAGE_SCN_MEM_EXECUTE flags on .data section, despite
__DATA segment not having anything alike. This breaks
memory protection of the resulting PE images.

REF: https://bugzilla.tianocore.org/show_bug.cgi?id=564
REF: https://bugzilla.tianocore.org/show_bug.cgi?id=2642

--- a/efitools/mtoc.c	2020-02-28 21:02:34.000000000 +0300
+++ b/efitools/mtoc.c	2020-04-01 15:58:02.000000000 +0300
@@ -821,9 +821,7 @@ struct arch *arch)
 		    scnhdrs[j].s_nlnno = 0;
 		    scnhdrs[j].s_flags = IMAGE_SCN_MEM_READ |
 					 IMAGE_SCN_MEM_WRITE |
-					 IMAGE_SCN_CNT_CODE |
-				         IMAGE_SCN_CNT_INITIALIZED_DATA |
-					 IMAGE_SCN_MEM_EXECUTE;
+					 IMAGE_SCN_CNT_INITIALIZED_DATA;
 		    scn_contents[j] = object_addr + sg->fileoff;
 		    j++;
 #ifdef HACK_TO_MATCH_TEST_CASE
@@ -1168,9 +1166,7 @@ struct arch *arch)
 		    scnhdrs[j].s_nlnno = 0;
 		    scnhdrs[j].s_flags = IMAGE_SCN_MEM_READ |
 					 IMAGE_SCN_MEM_WRITE |
-					 IMAGE_SCN_CNT_CODE |
-				         IMAGE_SCN_CNT_INITIALIZED_DATA |
-					 IMAGE_SCN_MEM_EXECUTE;
+					 IMAGE_SCN_CNT_INITIALIZED_DATA;
 		    scn_contents[j] = object_addr + sg64->fileoff;
 		    j++;
 		}
